### PR TITLE
Fix some IResources being leaked

### PR DESCRIPTION
**Pull request by ichttt**
This can be observed when setting the logger to DEBUG level in the log4j xml config, as minecraft replaces the inputstream with a leak tracking input stream when their logger has debug enabled.

### CODE REVIEW

File: src/main/java/net/minecraftforge/client/model/ModelDynBucket.java [hunk 1]
```diff
@@ -346,25 +347,34 @@ public boolean load(@Nonnull IResourceManager manager, @Nonnull ResourceLocation
...
-                    alphaMask.getIconWidth() == width && alphaMask.getIconHeight() == height)
+            IResource empty = null;
+            IResource mask = null;
+            try
             {
-                final int[][] oldPixels = sprite.getFrameTextureData(0);
-                final int[][] alphaPixels = alphaMask.getFrameTextureData(0);
-
-                for (int p = 0; p < width * height; p++)
+                empty = getResource(new ResourceLocation("textures/items/bucket_empty.png"));
```
Comment (mezz) [inline src/main/java/net/minecraftforge/client/model/ModelDynBucket.java]: Please use try-with-resources for these instead, it looks like this:
```java
try (
    IResource empty = getResource(new ResourceLocation("textures/items/bucket_empty.png"));
    IResource mask = getResource(new ResourceLocation(ForgeVersion.MOD_ID, "textures/items/vanilla_bucket_cover_mask.png"))
) {
    // do stuff
}
catch (IOException e)
{
    FMLLog.log.error("Failed to close resource. {}", e);
}
```

**Review by mezz — CHANGES_REQUESTED** [review]
Looks good, I saw this too when messing with the logs.
Please update these to use try-with-resources.

**Comment by ichttt** [comment]
@mezz I switched everything but the B3D and OBJ over to try-with-resources, because B3D and OBJ try to search in alternative places if the first resource lookup fails

**Comment by mezz** [comment]
Cool, looks good to me.

**Review by mezz — APPROVED** [review]

**Pull request merged**